Bytes32 hash: non-portable cast causes build failure on macOS

## Summary

Building on macOS with Apple Clang fails in `src/Bytes32.h` at:

```cpp
uint64_t *p = (size_t*)&b.buf;
```

This is a portability bug: the failure shows up on macOS (Apple Clang, arm64) but the underlying issue is the cast, not the OS alone.

## Environment

- OS: macOS (arm64)
- Compiler: Apple Clang (`/usr/bin/clang++`)
- Build flow: `make setup-golpe && make -j4`

## Reproduction

1. `git clone https://github.com/hoytech/strfry && cd strfry`
2. `git submodule update --init --recursive`
3. `make setup-golpe`
4. `make -j4`

## Error

```text
src/Bytes32.h:51:23: error: cannot initialize a variable of type 'uint64_t *' (aka 'unsigned long long *') with an rvalue of type 'size_t *' (aka 'unsigned long *')
```

## Notes

- The code casts `size_t*` to `uint64_t*`, assuming those pointer types can be used interchangeably for the same storage.
- On macOS (arm64) with Apple Clang, `size_t` is `unsigned long` and `uint64_t` is `unsigned long long`, so the conversion is ill-formed and the build fails.
- The same pattern may compile on some other toolchains where the typedef relationship happens to allow it, but it is not portable across platforms or guaranteed by the standard.
- Treating the buffer as `uint64_t` chunks via a `size_t*` cast is type-punning by pointer; a byte-oriented copy avoids that assumption.

## Suggested fix / direction

Use a `memcpy`-based approach:

```cpp
uint64_t words[4];
::memcpy(words, b.buf, sizeof(words));
return size_t(words[0] ^ words[1] ^ words[2] ^ words[3]);
```

This avoids the `size_t*` / `uint64_t*` mismatch and keeps the XOR hash over four 64-bit lanes.

--- a/src/Bytes32.h
+++ b/src/Bytes32.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <cstring>
 #include <string>
 #include <string_view>
@@ -47,8 +48,9 @@ namespace std {
     template<> struct hash<Bytes32> {
         std::size_t operator()(Bytes32 const &b) const {
             static_assert(sizeof(b.buf) == 32);
-            uint64_t *p = (size_t*)&b.buf;
-            return size_t(p[0] ^ p[1] ^ p[2] ^ p[3]);
+            uint64_t words[4];
+            ::memcpy(words, b.buf, sizeof(words));
+            return size_t(words[0] ^ words[1] ^ words[2] ^ words[3]);
         }
     };
 }

